### DISCUSSION TITLE
Frigate on Home Assistant

### DISCUSSION THREAD

**Discussion by leonbrag** [blakeblackshear/frigate#3015] [General]
Home Assistant offers two recommended ways to install (https://www.home-assistant.io/installation/):
1. OS running on bare metal
2. Containers

Question:
1. If Home Assistant is installed in Container (#2), and Frigate add-on added, is still a recommended way to run Frigate with TPUs, right? 
2. What is functionality is lost if Frigate is installed as a standalone, NOT as Home Assistant add-on? Does such installation just means pulling Frigate container (ex: blakeblackshear/frigate:stable-amd64)

thanks!

**Comment by NickM-27** [comment]
HomeAssistant addons are just docker containers that the HA OS runs. As far as the frigate container is concerned, besides some slight differences in setting up, things are identical. 

To answer your questions:
1. Yes, frigate in a container beside HA container is a good way to run it and won't be different functionally compared to other setups.
2. None, it is the exact same container in either case.

**Comment by RichardUUU** [comment]
The Frigate installation page says (twice): "If possible, it is recommended to run Frigate standalone in Docker and use [Frigate's Proxy Addon)"

I have Home Assistant installed as a VM on my unRaid machine. 

When Frigate is installed as a Home Assistant add-on, the only storage available for recordings is the storage inside my Home Assistant VM which is limited. For that reason, I installed Frigate as in a docker on my unRaid machine which gives me access to unRaid storage.

The Frigate Proxy Add-on integrates Frigate with Home Assistant, however there is a tradeoff. Apparently the Proxy Add-on uses the RTMP stream instead of RTSP used when Frigate is installed as a HASS Add-on (see [here](https://community.home-assistant.io/t/add-on-rtsptoweb-and-rtsptowebrtc/387846/104?u=richardu) )

That prevents you from using the RTSPtoWeb Add-on which can improve the lag when viewing live camera streams.

**Reply by NickM-27** [reply]
No version of frigate currently uses RTSP no matter its installation method. This has been talked about being changed but currently to use rtsptowebrtc the original camera rtsp must be used

**Reply by RichardUUU** [reply]
Good to know. Thanks. I assume it's still true that Frigate is unable to access NAS storage when installed as an add-on to a HASS VM. Is that also an issue when Frigate is installed as an add-on to a HASS docker?

**Reply by NickM-27** [reply]
> Good to know. Thanks. I assume it's still true that Frigate is unable to access NAS storage when installed as an add-on to a HASS VM. Is that also an issue when Frigate is installed as an add-on to a HASS docker?

I use Unraid and have both Home Assistant and frigate in a docker so frigate can have it's media mapped to whichever storage volume I want. If HA is a supervised docker with addons then it'll likely be limited to whichever volumes the HA docker is using

**Reply by blakeblackshear** [reply]
The proxy addon is just a simple nginx proxy. It doesn't run any part of frigate locally. It works even if frigate is run on a separate machine.

**Reply by RichardUUU** [reply]
I experienced a lag to streams that are pulled in from Frigate through the proxy add-on. That lag is greatly reduced by pulling streams directly from the cameras and using the [RTSPtoWeb](https://community.home-assistant.io/t/add-on-rtsptoweb-and-rtsptowebrtc/387846) add-on
```
camera:
  - platform: generic
```

**Reply by NickM-27** [reply]
> I experienced a lag to streams that are pulled in from Frigate through the proxy add-on. That lag is greatly reduced by pulling streams directly from the cameras and using the [RTSPtoWeb](https://community.home-assistant.io/t/add-on-rtsptoweb-and-rtsptowebrtc/387846) add-on
> 
> ```
> 
> camera:
> 
>   - platform: generic
> 
> ```

I'm assuming you mean Frigate Integration as the proxy addon is just a way to have the frigate UI within HA (like blakeblackshear said). 

That being said, that is expected. WebRTC is lower latency on top of the higher latency by bouncing between other protocols. I believe using rtsp-simple-server for rtsp to replace rtmp is on the feature roadmap.
